pkg/metrics/cluster/config_watcher.go

```diff
@@ -115,6 +116,24 @@ func (w *configWatcher) run(ctx context.Context) {
 	}
 }
 
+// nextReshard returns a channel to that will fill a value when the reshard
+// interval has elapsed.
+func (w *configWatcher) nextReshard(lastReshard time.Time) <-chan time.Time {
+	w.mut.Lock()
+	nextReshard := lastReshard.Add(w.cfg.ReshardInterval)
+	w.mut.Unlock()
+
+	now := time.Now()
+	if now.After(nextReshard) || now.Equal(nextReshard) {
+		ch := make(chan time.Time, 1)
+		ch <- now
+		return ch
+	}
+
+	remaining := time.Until(nextReshard)
```

Comment:
Given some of our issues do you think having some additional logging here on remaining time would be good.